tests/lib/utils

```diff
@@ -53,6 +53,23 @@ function improver_compare_output {
   [[ "$output" =~ "are identical." ]]
 }
 
+function improver_compare_output_lower_precision {
+  # Run nccmp to compare the output and known good output.
+  # Specify a tolerance to compare the output and known good output with
+  # lower precision.
+  # nccmp options:
+  #    -d means compare data
+  #    -m also compare metadata
+  #    -N ignore NaN comparisons
+  #    -s report 'Files X and Y are identical' if they really are.
+  #    -g compares global attributes in the file
+  #    -t defines a tolerance in terms of the absolute difference.
+
+  run nccmp -dmNsg -t 1e-3 "$1" "$2"
```

Comment:
Use of tolerance seems like it might be more widely applicable.  I'd like to discuss this with @benfitzpatrick , as improver_compare_output might benefit from eg a 1e-6 tolerance level.

For this particular function / EMOS coefficients, @arh89 and I agree the same tolerance should be used as in the unit tests: currently 1e-4.

Reply:
Please see the associated notebook PR for a review of precision level queries.

Reply:
I'm happy with using a sensible relative tolerance in the CLI tests by default for machine independence. An absolute one has issues.

Why does it give different results on the same machines - or doesn't it?

Reply:
Just to confirm, on the same machine the results are reproducible. The use of a tolerance here is so that the results will match when produced on different machines.

Reply:
I've added a comment to this test to clarify that the tolerance chosen for this function is specifically for ensemble calibration.

Reply:
Thanks.